### PR TITLE
feat: csp nonce generation

### PR DESCRIPTION
**Pull request by climba03003**
Resolve: #89 

This PR aims to simplify the usage for csp nonce generation with `fastify` and `helmet`.

Changes:
- introduce new option `enableCSPNonces`
- decorate `FastifyReply` with `cspNonce`

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by climba03003** [comment]
Opps, missed this comment https://github.com/fastify/fastify-helmet/pull/90#issuecomment-677647961
I think this is more like the `fastify` style of `csp nonce`.

File: README.md [hunk 1]
```diff
@@ -32,6 +32,37 @@ fastify.listen(3000, err => {
 })
 ```
 
+### Content-Security-Policy Nonce
+
```
Comment (mcollina) [inline README.md:36]: Let's document some of this is implemented in this module instead to provide the decorated values as decorators in the reply.

File: index.js [hunk 2]
```diff
@@ -2,14 +2,47 @@
...
+      // allow both script-src or scriptSrc syntax
+      const scriptKey = Array.isArray(cspDirectives['script-src']) ? 'script-src' : 'scriptSrc'
+      cspDirectives[scriptKey] = Array.isArray(cspDirectives.scriptSrc) ? cspDirectives.scriptSrc : []
+      cspDirectives[scriptKey].push('nonce-' + reply.cspNonce.script)
+      // allow both style-src or styleSrc syntax
+      const styleKey = Array.isArray(cspDirectives['style-src']) ? 'style-src' : 'styleSrc'
+      cspDirectives[styleKey] = Array.isArray(cspDirectives.styleSrc) ? cspDirectives.styleSrc : []
+      cspDirectives[styleKey].push('nonce-' + reply.cspNonce.style)
+
+      const cspMiddleware = helmet.contentSecurityPolicy({ directives: cspDirectives, reportOnly: cspReportOnly })
```
Comment (mcollina) [inline index.js:41]: Please move the generation of this middleware outside of the hook.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: index.js (hunk 2 above)
Comment (climba03003) [reply index.js:41]: I think it is a must for this middleware to exist inside the hook. As we are not passing the `function` inside helmet. 
The options for `helmet.contentSecurityPolicy` is unique for every calls.

File: README.md (hunk 1 above)
Comment (climba03003) [reply README.md:36]: Would you mind to explain more? Should I elaborate more on how this options works?

Comment (mcollina) [reply README.md:36]: This package is called `-helmet` as it wraps helmet itself. This feature is partially implemented inside this module, we should list the differences.

Comment (climba03003) [reply README.md:36]: Update.
I also add an example about how to use helmet option only.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
Go ahead and merge & release this!

**Pull request merged**